op-devstack/dsl/l2_el.go

```diff
@@ -28,3 +34,48 @@ func (el *L2ELNode) String() string {
 func (el *L2ELNode) Escape() stack.L2ELNode {
 	return el.inner
 }
+
+func (el *L2ELNode) BlockRefByLabel(label eth.BlockLabel) eth.BlockRef {
+	ctx, cancel := context.WithTimeout(el.ctx, DefaultTimeout)
+	defer cancel()
+	block, err := el.inner.EthClient().BlockRefByLabel(ctx, label)
+	el.require.NoError(err, "block not found using block label")
+	return block
+}
+
+func (el *L2ELNode) Advance(label eth.BlockLabel, block uint64) CheckFunc {
+	return func() error {
+		initial := el.BlockRefByLabel(eth.Unsafe)
+		target := initial.Number + block
+		el.log.Info("expecting chain to advance", "chain", el.inner.ChainID(), "label", label, "target", target)
+		attempts := int(block + 3) // intentionally allow few more attempts for avoid flaking
+		return retry.Do0(el.ctx, attempts, &retry.FixedStrategy{Dur: 2 * time.Second},
+			func() error {
+				unsafeHead := el.BlockRefByLabel(eth.Unsafe)
+				if unsafeHead.Number >= target {
```

Comment:
```suggestion
				unsafeHead := el.BlockRefByLabel(label)
				if unsafeHead.Number >= target {
```
And you may want to rename `unsafeHead` to something more generic.

Reply:
https://github.com/ethereum-optimism/optimism/pull/15816/commits/ff9cd6e90784bf34a4f90e1c882043f415713749